freqtrade/strategy/backtest_lookahead_bias_checker.py

```diff
@@ -0,0 +1,252 @@
+import copy
+import pathlib
+import shutil
+from copy import deepcopy
+from datetime import datetime, timedelta, timezone
+
+from pandas import DataFrame
+
+from freqtrade.configuration import TimeRange
+from freqtrade.data.history import get_timerange
+from freqtrade.exchange import timeframe_to_minutes
+from freqtrade.optimize.backtesting import Backtesting
+
+
+class VarHolder:
+    timerange: TimeRange
+    data: DataFrame
+    indicators: DataFrame
+    result: DataFrame
+    compared: DataFrame
+    from_dt: datetime
+    to_dt: datetime
+    compared_dt: datetime
+
+
+class Analysis:
+    def __init__(self):
+        self.total_signals = 0
+        self.false_entry_signals = 0
+        self.false_exit_signals = 0
+        self.false_indicators = []
+        self.has_bias = False
+
+    total_signals: int
+    false_entry_signals: int
+    false_exit_signals: int
+
+    false_indicators: list
+    has_bias: bool
+
+
+class BacktestLookaheadBiasChecker:
+
+    def __init__(self):
+        self.exportfilename = None
+        self.strategy_obj = None
+        self.current_analysis = None
+        self.local_config = None
+        self.full_varHolder = None
+
+        self.entry_varHolder = None
+        self.exit_varHolder = None
+        self.entry_varHolders = []
+        self.exit_varHolders = []
+        self.backtesting = None
+        self.minimum_trade_amount = None
+        self.targeted_trade_amount = None
+        self.failed_bias_check = True
```

Comment:
those variables get put there since pycharm else warns me about it 
"instance attribute defined outside __init__"

can remove it if you like, please advise

Reply:
well i'm not necessarily against these lines - but about the usage pattern this creates (this, in combination with whee they're overwritten).

i think you should pass config into the constructor - and fill these variables properly (sorry if that didn't come across ...).

Reply:
done